zebra/zebra_ptm.c

```diff
@@ -642,6 +642,7 @@ int zebra_ptm_sock_read(struct thread *thread)
 	do {
 		rc = ptm_lib_process_msg(ptm_hdl, sock, ptm_cb.in_data,
 					 ZEBRA_PTM_MAX_SOCKBUF, NULL);
+		memset(ptm_cb.in_data, 0, ZEBRA_PTM_MAX_SOCKBUF);
```

Comment:
I disagree that this actually fixes the issue with decoding.  This just prevents this particular overflow you have found.  Not others.  We must fix the actual decoding of the messages to not do the wrong thing when reading data from the wire.  As that it implies then I could just send incorrect data and cause bad issues.

Additionally you added a stack decode to the PR not to the commit message.  If that stack decode is important to understanding the issue it belongs in the commit message.  Please update

Reply:
Yes Donald, It only fixes the overflow problem. Sure will update the commit message.

Reply:
This is not the correct fix then.  We must properly decode with limits on length of buffer read as well as proper parsing of data from the input.

Reply:
ok, I am moving this PR in draft mode and update the code.
Thanks for review.